### PR TITLE
fix: mode behaviour

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

fix

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes

**Summary**
Fix mode priority 
- mode form flag
- config
- NODE_ENV

**Does this PR introduce a breaking change?**
Nay

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/groups/resolveMode.js [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
+
+    if (mode) {
+        finalMode = mode;
+    } else if (configMode) {
+        finalMode = configMode;
+    } else if (NODE_ENV && (NODE_ENV === PRODUCTION || NODE_ENV === DEVELOPMENT)) {
         finalMode = NODE_ENV;
     } else {
-        finalMode = mode || PRODUCTION;
+        finalMode = PRODUCTION;
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/resolveMode.js:27]: No need this, webpack do it itself (default is production)

**Review by evilebottnawi — COMMENTED** [review]
Also can you add multicompiler tests

File: packages/webpack-cli/lib/groups/resolveMode.js (hunk 1 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/groups/resolveMode.js:27]: gotcha

**Comment by anshumanv** [comment]
> Also can you add multicompiler tests

`-c 1.js -c 2.js`

which config mode should be priority? @evilebottnawi

**Comment by evilebottnawi** [comment]
With `--mode=production` both configurations should set `mode: production`, without `--mode=production` but with `mode: 'production'` in config should keep values, with `NODE_ENV` and without `mode: 'production'` should use `NODE_ENV`, same logic, maybe better to use `module: development` to test, because default value is `production`

File: packages/webpack-cli/lib/groups/resolveMode.js (hunk 1 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/groups/resolveMode.js:29]: Need this because https://github.com/webpack/webpack-cli/issues/1678 hence added

**Comment by anshumanv** [comment]
> With --mode=production both configurations should set mode: production, without --mode=production but with mode: 'production' in config should keep values, with NODE_ENV and without mode: 'production' should use NODE_ENV, same logic, maybe better to use module: development to test, because default value is production

Needed more work than I thought, looks good now

File: packages/webpack-cli/lib/groups/resolveMode.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/resolveMode.js:29]: Okay, let's refactor it in future

**Review by evilebottnawi — APPROVED** [review]

File: packages/webpack-cli/lib/groups/resolveMode.js (hunk 1 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/groups/resolveMode.js:29]: Sure sure

**Review by evilebottnawi — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Pull request merged**